packages/mobile/src/home/CashInBottomSheet.tsx

```diff
@@ -26,13 +43,66 @@ function CashInBottomSheet() {
     setModalVisible(false)
   }
 
+  const asyncRampInfo = useAsync(
+    async () => {
+      if (!account) {
+        Logger.error(TAG, 'No account set')
+        return
+      }
+      // Use cEUR if that is their local currency, otherwise default to cUSD
+      const currencyToBuy =
+        localCurrency === LocalCurrencyCode.EUR ? CiCoCurrency.CEUR : CiCoCurrency.CUSD
+
+      try {
+        const providers = await fetchProviders({
+          userLocation,
+          walletAddress: account,
+          fiatCurrency: localCurrency,
+          digitalAsset: currencyToBuy,
+          fiatAmount: 20,
+          digitalAssetAmount: 20,
+          txType: 'buy',
+        })
+        const rampProvider = providers?.find((provider) => provider.name === 'Ramp')
+        const rampAvailable = !!(
+          rampProvider &&
+          rampProvider?.cashIn &&
+          !rampProvider.restricted &&
+          !rampProvider.unavailable
+        )
+        return {
+          rampAvailable,
+          rampURL: rampProvider?.url,
+        }
+      } catch (error) {
+        Logger.error(TAG, 'Failed to fetch CICO providers')
+      }
+    },
+    [],
+    {
+      onSuccess: () => setModalVisible(true),
+      onError: () => setModalVisible(true),
```

Comment:
is this intentional? setting visible to true on error?

Reply:
Yep! This async function is solely for getting the ramp availability info. If it errors out, there's no reason we can't still show the normal cash in button